src/main/scala/firrtl/transforms/LegalizeClocksAndAsyncReset.scala

```diff
@@ -33,19 +33,30 @@ object LegalizeClocksTransform {
   def onStmt(namespace: => Namespace)(stmt: Statement): Statement =
     stmt.map(onStmt(namespace)) match {
       // Proper union types would deduplicate this code
-      case r: DefRegister if illegalClockExpr(r.clock) =>
-        val node = DefNode(r.info, namespace.newTemp, r.clock)
-        val rx = r.copy(clock = WRef(node))
-        Block(Seq(node, rx))
-      case p: Print if illegalClockExpr(p.clk) =>
+      case r: DefRegister if (isLiteralExpression(r.clock) || isLiteralExpression(r.reset)) =>
+        val (clockNodeOpt, rxClock) = if (isLiteralExpression(r.clock)) {
+          val node = DefNode(r.info, namespace.newTemp, r.clock)
+          (Some(node), r.copy(clock = WRef(node)))
+        } else {
+          (None, r)
+        }
+        val (resetNodeOpt, rx) = if (isLiteralExpression(r.reset)) {
+          val castToAsync = DoPrim(PrimOps.AsAsyncReset, Seq(r.reset), Nil, AsyncResetType)
+          val node = DefNode(r.info, namespace.newTemp, castToAsync)
```

Comment:
Are we sure that the cast isn't already there and that the emitter isn't just dropping it?

Reply:
I added this because a downstream pass complains that sync reset should have been removed, if i just carry the expression forward into the node. I guess the more correct thing to do is check if theres a cast, and add it if it doesn't already exist.

Reply:
I see. I'll investigate a bit.